### ISSUE TITLE
Domain Verification API

### ISSUE THREAD

**Issue by Shachindra** [NetSepio/gateway#44]
Need to integrate a feature where an authenticated user can verify a domain ownership on the NetSepio Webapp. In order to do this we need them to create a TXT Record for the domain with a value given by us and then we can confirm this at our backend. @huisq Let me know if I missed anything.

Required APIs:

1. Add a Domain- /api/v1.0/domain [POST]
        - Req: domainName (netsepio.com)
        - Res: payload: {domainId: uuid, domainName: string, txRecord: {key, value}, admins: [], verified: true/false}
2. Read Domain by ID - /api/v1.0/domain [GET] (With Query Paramater: domainId)
3. Verify a Domain by ID - /api/v1.0/domain/verify [PATCH]
         - Logic: Use nslookup; Run this example ```nslookup -q=txt netsepio.com```
         - Res: payload: {domainId: uuid, domainName: string, txRecord: {key, value}, admins: [], verified: true/false}
4. Remove a Domain by ID [DELETE]
5. Update Admins to Domain by ID: /api/v1.0/domain/admin [PATCH]
        - Req: Array of userIDs (Can't be Blank - must have at least 1 userid in the array)
        - Res: payload: {domainId: uuid, domainName: string, txRecord: {key, value}, admins: [], verified: true/false}
6. Search a Domain by Name - [GET] (With Query Paramater: domainName)
7. List all domains - /api/v1.0/domain [GET] Response - Array of domains registered with pagination

**Comment by huisq** [comment]
Do you need this feature to run on smart contract as well? another function and/or storage? If so, could add another role title and use existing role functions

**Comment by Shachindra** [comment]
I'm not sure what advantage would the domain verification on the blockchain bring except that the badge/reputation that comes with it. We could:
1. Build a simple NFT contract that issues the Domain NFT to the verifier address which serves as a Badge/Achievement under profile section (this should not affect OR connected to our existing smart contract)
2. Push this on-chain feature as part of Milestone 2 with proper due-diligence.

wdyt?

**Comment by huisq** [comment]
Awesome. Just clicked in my head how this NFT badge could work as a verification and perhaps even almost a reward system in future. Let's go with that.

**Comment by Shachindra** [comment]
Yes, We'll have multiple NFT collections to issue various achievements to users for gamification and rewards purposes. We'll target the onchain integration during milestone 2. For now @thisisommore will work on the APIs...

**Comment by thisisommore** [comment]
@Shachindra 6 and 7 can be merged?

**Comment by thisisommore** [comment]
@Shachindra I am not aware of RTF much, what will its type in golang, will it required any validation, can you provide any resources?

**Comment by Shachindra** [comment]
> @Shachindra 6 and 7 can be merged?

Yes.

**Comment by Shachindra** [comment]
> @Shachindra I am not aware of RTF much, what will its type in golang, will it required any validation, can you provide any resources?

It will store the data coming from the embed taken from an editor (like https://summernote.org/)

**Comment by thisisommore** [comment]
@Shachindra Should I merge 2 with current query api #98

**Comment by Shachindra** [comment]
Sure
